src/interactivity/win32/ConsoleControl.cpp

```diff
@@ -13,31 +13,51 @@
 
 using namespace Microsoft::Console::Interactivity::Win32;
 
-#pragma region IConsoleControl Members
+#ifdef CON_USERPRIVAPI_INDIRECT
+ConsoleControl::ConsoleControl()
+{
+    // NOTE: GetModuleHandleW is rather expensive, but GetProcAddress is quite cheap.
+    const auto user32 = GetModuleHandleW(L"user32.dll");
+    _consoleControl = reinterpret_cast<PfnConsoleControl>(GetProcAddress(user32, "ConsoleControl"));

```

Comment:
so ok. this might be dumb, but what if you got rid of all the indirect checks *below* the constructor. and did this instead.

```c++
ConsoleControl::ConsoleControl()
{
#ifdef CON_USERPRIVAPI_INDIRECT
// load user32
_consoleControl = Get();
_enterReaderModeHelper = Get();
#else
_consoleControl = &::ConsoleControl;
#endif
}
```

then it's all indirect. but like. the code is simpler?

Reply:
I wouldn't do this. Function pointers are IMO an inconvenience when debugging in disassembly and they're subject to CFG.